static/app/utils/useRouteContext.tsx

```diff
@@ -0,0 +1,23 @@
+import {useContext} from 'react';
+
+import {RouteContext} from 'sentry/views/routeContext';
+
+function getCallerName() {
```

Comment:
Nice idea, but I think it's an overkill for this PR. The reasoning is that you already have the stack trace when you throw and it's easy enough to see which function this originated from (the component frame is usually not more than a few stack frames away) and because stack traces also have different formats in different browsers, so you'd need to make sure you properly account and support that by adding tests. I wouldnt want us to end up with ` called outside of routes provider`. 

For the purpose of this PR, I would just throw, but I would encourage you to create a separate PR with a helper that implements and tests this and we can continue the discussion there and see if we want to adopt this in all of the rest of our contexts